05_Datathon/diabetes_prediction_example.Rmd

```diff
@@ -0,0 +1,336 @@
+
+
+### 5.3 Predict diabetes mellitus in five unseen patients from test set
+
+#### Select unseen patients from testing set
+
+- Let's look five unseen patients in our test set.
+- What outcome do you expect for these patients?
+
+```{r select_unseen, echo=TRUE}
+unseen <- head(test_set, 5) %>% 
+          select(age, bmi, gender, glucose_apache)
+
+print(unseen)
+```
+
+#### Predict the diabetes mellitus in unseen patients
+
+- Is the model correct for this patient?
+- `type="prob"` to output probabilities
+- `type="class"` to output classes
+
+```{r predict_unseen, echo=TRUE}
+# Make predictions on testing set
+forest_pred <- predict(forest,
+                     newdata = test_set,
+                     type="class")
+# Combine unseen patients data with corresponding predictions
+data.frame(age = unseen$age,bmi = unseen$bmi,gender = unseen$gender, glucose_apache = unseen$glucose_apache, prediction = forest_pred[1:5], truth_value = test_set$target_variable[1:5]) %>% kable()
+```
+
+## 6. Model Evaluation
+
+### 6.1 Creating our confusion matrix
+
+- A confusion matrix relates predictions to the ground truth.
+- Forms the basis for evaluation metrics.
+- Alive is our "0" (-ve). Expired is our "1" (+ve).
+
+```{r create_confusion_matrix, echo=TRUE}
+library("caret")
+
+# Note that "forest refers to the model trained previously
+forest_preds = predict(forest, 
+                     newdata = test_set,
+                     type = "class")
+
+# Ground truth is recorded in the GOSSIS data
+confusionMatrix(forest_preds,
+                test_set$target_variable,
+                positive='1')
+
+```
+
+### 6.2 Evaluation Measures
+
+#### 6.2.1 Accuracy
+
+- How many shots did we take? What proportion hit the target?
+- 106 shots. 96 hit the mark. Accuracy = ?
+- Accuracy = $\frac{(TP + TN)}{(TP + TN + FP + FN)}$
+
+#### 6.2.2 Sensitivity
+
+- (AKA "Recall" AKA "True Positive Rate")
+- Of the patients who do not survive, what proportion did we correctly predict?
+- 9 people did not survive. We called 2. Sensitivity = ?
+- Recall = $\frac{(TP)}{(TP + FN)}$
+
+#### 6.2.1 Specificity 
+
+- (AKA "True Negative Rate")
+- What proportion of survivors did we say would survive?
+- 97 survivors. We predicted 5 would die. Specificity = ?
+- Specificity = $\frac{(TN)}{(TN + FP)}$
+
+#### 6.2.1 Area under ROC Curve (AUC)
+
+- Developed in the 1940s by radar operators.
+- Popular measure of discrimination.
+- Plots *1 - specificity* vs. *sensitivity* at varying probability thresholds.
+- 0.5 is terrible. 1.0 is the dream.
+- AUC of 0.9 tells us that the 90% of time our model will assign a higher risk to a randomly selected patient with an event than to a randomly selected patient without an event
+
+We'll use the [ROCR package](https://cran.r-project.org/web/packages/ROCR/index.html) ("Visualizing the Performance of Scoring Classifiers") to plot the curve.
+
+```{r AUROC, echo=TRUE}
+library("ROCR")
```

Comment:
Up to you.  i used ROCR for a long time, but now use pROC and ModelMetrics:

```{r}
my_roc <-  pROC::roc(test_set$target_variable,forest_probs,ci=TRUE)
print(my_roc)
plot(my_roc)
```

seems more intuitive and computes CIs.